modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -584,6 +584,10 @@ Error GDScriptAnalyzer::resolve_class_inheritance(GDScriptParser::ClassNode *p_c
 	return OK;
 }
 
+GDScriptParser::DataType GDScriptAnalyzer::resolve_container_datatype(int p_index, GDScriptParser::TypeNode *p_type) {
```

Comment:
Not sure why you added this function since it's only called once. Seems to me this can just be embedded in the logic for `resolve_datatype()` as this shouldn't be accessed elsewhere.

Reply:
I was thinking the explicit name would be more clear than some optional argument variable or utilizing ternary operators elsewhere. It's just the one use for now, but I would utilize this with typed dictionaries as well

Still, no harm in throwing together embedded logic to the existing function instead